torch/_inductor/codegen/triton.py

```diff
@@ -1646,7 +1650,21 @@ def tan(x):
     @staticmethod
     @maybe_upcast_float32()
     def tanh(x):
-        return f"libdevice.tanh({x})"
+        cse_var = V.kernel.cse.varname_map.get(x)
+        if cse_var and hasattr(cse_var, "dtype"):
+            dtype = cse_var.dtype
+        else:
+            dtype = None
```

Comment:
Did you see this branch happen? We probably want to skip fast tanhf if dtype is None.